daemon/prompter.c

```diff
@@ -726,12 +726,27 @@ prompter_prompt_invocation_cb(GObject *obj, GAsyncResult *res, gpointer aptr)
 static GVariant *
 prompter_invocation_args(const prompter_t *self, appinfo_t *appinfo)
 {
+    gchar *desktop                 = NULL;
+    const stringset_t *permissions = NULL;
+    stringset_t *filtered          = NULL;
+    GVariant *args                 = NULL;
+
     GVariantBuilder builder;
     g_variant_builder_init(&builder, G_VARIANT_TYPE("a{sas}"));
 
-    const stringset_t *permissions = appinfo_get_permissions(appinfo);
-    stringset_t *filtered = service_filter_permissions(prompter_service(self),
-                                                       permissions);
+    /* Prompting is applicable only to applications that have desktop
+     * file in standard directory.
+     *
+     * Applications that have desktop file only in sailjail specific
+     * directory are either allowed by default (in which case we should
+     * not end up here) or are denied without prompting.
+     */
+    desktop = path_from_desktop_name(appinfo_id(appinfo));
+    if( access(desktop, R_OK) == -1 )
+        goto EXIT;
```

Comment:
Note: g_variant_builder_init() has been called and goto skips g_variant_builder_end() -> AFAICT that is a potential memory leak / not nice even if it is not.

Doing the desktop check 1st would avoid that problem.